### ISSUE TITLE
need help regarding  training 

### ISSUE THREAD

**Issue by sarvang008** [karimrattani/rare_event_detection#7]
The pre-trained model which the project uses is the ucf-101 model but the dataset used for testing is ucf-crime, so does the ucf-101 model include the videos from ucf crime dataset?
Also, the user can train the model by himself can you help me how does training work as the videos do not contain the exact time where the anomaly exists how will the model know there is an anomaly?

**Comment by mtfuller** [comment]
Hey, sorry for late response.

>The pre-trained model which the project uses is the ucf-101 model but the dataset used for testing is ucf-crime, so does the ucf-101 model include the videos from ucf crime dataset?

No, the UCF-101 dataset was used to train the pre-trained C3D model for action detection. It does not contain any of the UCF-Crime videos. 

The authors of the paper (https://arxiv.org/pdf/1801.04264.pdf) ran the UCF-Crime videos through the pre-trained C3D model to get a vector of features. This vector was the input to train a few fully connected layers that would output a single normalized value (0.0=no anomaly, 1.0=has anomaly).

> Also, the user can train the model by himself can you help me...

In `dataset/data.csv`, you should find a list of .mp4 files with their corresponding anomaly label. You should be able to modify that .csv to include any video you wish to include in training/testing. 

**Note**: the data.csv in this repo only includes part of the UCF-Crime dataset, as our team didn't have time left to train on the full dataset.

Running `python main.py` should start the script that will automatically divide the dataset into a train and a test set. Let me know if you encounter any errors, and I may can help.

>...how does training work as the videos do not contain the exact time where the anomaly exists how will the model know there is an anomaly?

It's true that the video dataset does not mark the exact time an anomaly is present. We were trying to reproduce the results in a paper from UCF: https://arxiv.org/pdf/1801.04264.pdf.

In that paper, they claim that their model architecture was able learn when anomalies occurred, even if the entire video was labeled true or false. I was a bit skeptical.

Back in Spring, I really wish we would have been able to reproduce their results exactly. I tried reaching out to them a long time ago to see if they would be willing to share their Caffe/Tensorflow code, but I didn't get any response. 

I would suggest reaching out to them, if possible. I'm sure it would be helpful to at least be able to reproduce their results.
